uuid/uuid.go

```diff
@@ -72,3 +73,25 @@ func (u UUID) String() string {
 	}
 	return fmt.Sprintf(strFormat, b...)
 }
+
+// MarshalJSON implements the marshaller interface.
+// This allows us to actually read and edit the json file
+func (u *UUID) MarshalJSON() ([]byte, error) {
+	return []byte(`{"UUID" : "` + u.String() + `"}`), nil
+}
+
+// UnmarshalJSON implements the unmarshaller interface.
+// This allows us to actually read and edit the json file
+func (u *UUID) UnmarshalJSON(b []byte) error {
+	j := make(map[string]string)
+
+	if err := json.Unmarshal(b, &j); err != nil {
+		return err
+	}
+	g, err := Parse(j["UUID"])
+	if err != nil {
+		return err
+	}
+	copy(u[:], g[:])
```

Comment:
should you check that `len(g) == Size` before copying?

Reply:
both of them are UUIDs, the length is fixed since it's an array, not a slice.

Reply:
you're right, I didn't realize that